app/src/main/java/hypr/a255bits/com/hypr/DrawableImageView.kt

```diff
@@ -0,0 +1,102 @@
+package hypr.a255bits.com.hypr
+
+import android.content.Context
+import android.graphics.*
+import android.util.AttributeSet
+import android.view.MotionEvent
+import android.widget.ImageView
+
+
+open class DrawableImageView(context: Context?, attrs: AttributeSet?) : ImageView(context, attrs){
+
+    var bitmap: Bitmap? = null
+    val paint = Paint()
+    var scaledBitmap: Bitmap? = null
+    val oldFaceLocations = mutableListOf<Rect>()
+    private val faceLocation = mutableListOf<Rect>()
+    val touchPaint = Paint()
+    private var boundsListener: DrawableImageViewTouchInBoundsListener? = null
+    init{
+        touchPaint.color = Color.BLUE
+        touchPaint.style = Paint.Style.STROKE
+        touchPaint.strokeWidth = 1.5f
+    }
+
+
+    fun setBoundsTouchListener(boundsListener: DrawableImageViewTouchInBoundsListener){
+        this.boundsListener = boundsListener
+    }
+    override fun onMeasure(widthMeasureSpec: Int, heightMeasureSpec: Int) {
+        val parentWidth = MeasureSpec.getSize(widthMeasureSpec)
+        val parentHeight = MeasureSpec.getSize(heightMeasureSpec)
+        this.setMeasuredDimension(parentWidth, parentHeight)
+        super.onMeasure(widthMeasureSpec, heightMeasureSpec)
+    }
+
+    override fun onDraw(canvas: Canvas?) {
+        super.onDraw(canvas)
+        faceLocation.forEach { item ->
+            oldFaceLocations.add(Rect(item))
+        }
+        scaledBitmap = scaleBitmap(bitmap!!)
+        val centreX = (width - scaledBitmap!!.width) / 2
+        val centreY = (height - scaledBitmap!!.height) / 2
+        canvas?.drawBitmap(scaledBitmap, centreX.toFloat(), centreY.toFloat(), paint)
+        scaleTouchInputBoxesToImagePosition(scaledBitmap!!, centreX, centreY)
+    }
+
+    override fun onTouchEvent(event: MotionEvent?): Boolean {
+        faceLocation.forEachIndexed() { i, faceLocation ->
+            if(faceLocation.contains(event?.x!!.toInt(), event.y.toInt())){
+                bitmap?.let { boundsListener?.onBoundsTouch(it, i) }
+            }
+        }
+
+        return super.onTouchEvent(event)
+    }
+
+    private fun scaleTouchInputBoxesToImagePosition(bit: Bitmap, centreX: Int, centreY: Int){
+        faceLocation.forEach { rect ->
+            val widthDifference = bit.width.toFloat() / bitmap!!.width.toFloat()
+            val heightDifference = ((bit.height.toFloat()))/ (bitmap!!.height.toFloat())
+            rect.right = (rect.right * widthDifference).toInt()
+            rect.left = (rect.left * widthDifference).toInt()
+            rect.bottom = (rect.bottom * heightDifference).toInt()
+            rect.bottom += centreY
+            rect.top = (rect.top * heightDifference).toInt()
+            rect.top += centreY
```

Comment:
Does `centreX` need to be used here?

Reply:
CentreY is used because I centered the image in the center of the screen. It's originally at the top.